### DISCUSSION TITLE
Server disappearing

### DISCUSSION THREAD

**Discussion by kdimail** [flant/grafana-statusmap#283] [Q&A] [answered]
Hello Statusmap Plugin Team.
I faced into the strange behavior of Statusmap/Grafana.
If I have some problematic server, after period of time it disappears from the dashboard but AFTER fixing it I see timeline on the Statusmap in red (downtime) and again green color.
My question is: how can I configure Statusmap/Grafana/Prometheus (_I really did not find where is it_) for actual/real view and not "customized"? I mean - if I have some problem with the server I want to see it ALL TIME in red till resolved problem otherwise I do not know what incidents I have because I do not see them online. Thanks.

**Comment by aksemats** [comment, marked as answer]
Hello! Plugin can't display something without data. Let's find out if you need more queries or just change plugin settings. Please give more information: screenshot without failed server, screenshot with restored server, plugin settings, datasource queries and responses.

**Reply by kdimail** [reply]
Thanks for your response.
![statusmap_cache](https://user-images.githubusercontent.com/62828900/190080198-155fe754-8c1a-4ab4-a5c9-6ad5a8c6452f.jpg)
Here the picture.

**Reply by aksemats** [reply]
So you've got row for each server and there are 3 states: the whole row is green if server is OK, no row if server is failed and red/green row when server is up again. Is failed server's row dissapper right after failing? If that's the problem, I think it is more a query issue.
Please attach datasource queries with data responses to figure out data correctness and to reproduce this behaviour.

**Reply by kdimail** [reply]
Is failed server's row dissapper right after failing? If that's the problem? - Exactly! My data source is a Prometheus. Here the screenshot of PromQL query
![2022-09-14 10_18_01-Window](https://user-images.githubusercontent.com/62828900/190087252-dfc1f89b-1c35-488e-98ce-22215bd2269c.jpg)

**Reply by aksemats** [reply]
Ah, I see. You need queries for each status as stated in README. The query `up == 1` will return no data if Prometheus has all `0` for the dashboard range.

**Reply by kdimail** [reply]
You mean here?
![2022-09-14 10_18_01-Window](https://user-images.githubusercontent.com/62828900/190091137-51493b88-cd65-4d1b-8a36-0be4f0593a96.jpg)

**Reply by aksemats** [reply]
There is an example setup for Prometheus https://github.com/flant/grafana-statusmap#prometheus

**Reply by kdimail** [reply]
Unfortunately I really do not understand what exactly I should do. I've read (before) this README a couple of times but because of coffee maker example I do no see how to config in my case.

**Reply by aksemats** [reply]
That example uses advanced techniques to get correct data from Prometheus (max_over_time, count, record rules). It is better to use them to eliminate false positives and data loss.
The main idea is to create queries to have 0 and 1 for each possible status (0 and 1 for status 0, 0 and 1 for status 1, 0 and 1 for status 2, and so on).

In case you just explore a new plugin I think you can use `up == bool 1` for simplicity. This query returns 1 for timestamps when `up` is 1 and 0 for timestamps when `up` is not 1. See https://prometheus.io/docs/prometheus/latest/querying/operators/#comparison-binary-operators

**Reply by kdimail** [reply]
Thank you so much for your patience and answer! 
I've been searching for weeks (!) for the solving my problem.
Have a perfect day!

**Comment by kdimail** [comment]
Hello. I have a question: how can I reduce to server name only ALL the description that Statusmap takes from the Prometheus config file. Is there some custom display templates or something like that? Thanks.
![2022-09-15 15_59_23-Window](https://user-images.githubusercontent.com/62828900/190410349-8de0e745-6c08-4281-b662-4ea5c365e253.jpg)

**Reply by diafour** [reply]
Statusmap takes nothing from the Prometheus configuration. Please, create new discussion with more details.
